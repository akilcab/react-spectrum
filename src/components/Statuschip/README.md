# Description of component

StatusChip is an element that represents an input, an attribute or an action.

#### Example :

![](assets/desc.png)

## How to import

```node
import { StatusChip } from "app/shared/components/StatusChip";
```

## How to use

````node
....
    <StatusChip size="small" label="CONNECTED" variant="outlined" />
....
````

## Props

| Props | Required | Description | Type |
| :---- | :------- | :---------- | :--- |
| `children` | true | The content of the button. | `node` |
| `color` | false | Allows you to define the foreground color of a text element and its possible decorations. | `string` |
| `label` | false | The content of the label. | `node` |
| `background` | false | Allows you to define the different values of the properties related to the management of the backgrounds of an element (color, image, origin, size, repetition, etc. | `string` |

## CSS properties we can override

You can override all CSS properties of the ChipProps type of Material UI.
