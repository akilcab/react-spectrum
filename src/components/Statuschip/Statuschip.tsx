import React from "react";
import { Theme, withStyles } from "@material-ui/core/styles";
import { Chip } from "@material-ui/core";
import { STATUS } from "./constants/status";

export interface StatuschipProps {
  color?: string;
  label?: string;
  background?: string;
  children?: React.ReactNode;
  status?: string;
}

export const Statuschip = withStyles((theme: Theme) => ({
  root: {
    background: (props: StatuschipProps) => {
      const { label } = props;
      if (label === STATUS.IN_PROGESS || label === STATUS.CUSTOMIZED) {
        return "#f6efe7";
      }
      if (label === STATUS.CANCELLED) {
        return "#ffefed";
      }

      if (
        label === STATUS.COMPLETED ||
        label === STATUS.RECLAIMED ||
        label === STATUS.CONNECTED
      ) {
        return "#ebf3f1";
      }

      return "initial";
    },
    border: (props: StatuschipProps) => {
      const { label } = props;
      if (label === STATUS.IN_PROGESS || label === STATUS.CUSTOMIZED) {
        return `1px solid ${theme.palette.warning.light}`;
      }
      if (label === STATUS.CANCELLED) {
        return `1px solid ${theme.palette.error.light}`;
      }

      if (
        label === STATUS.COMPLETED ||
        label === STATUS.RECLAIMED ||
        label === STATUS.CONNECTED
      ) {
        return `1px solid #3ca896`;
      }

      return "none";
    },
    color: (props: StatuschipProps) => {
      const { label } = props;
      if (label === STATUS.IN_PROGESS || label === STATUS.CUSTOMIZED) {
        return theme.palette.warning.light;
      }
      if (label === STATUS.CANCELLED) {
        return theme.palette.error.light;
      }

      if (
        label === STATUS.COMPLETED ||
        label === STATUS.RECLAIMED ||
        label === STATUS.CONNECTED
      ) {
        return "#3ca896";
      }

      return "initial";
    },
  },
}))(Chip);
