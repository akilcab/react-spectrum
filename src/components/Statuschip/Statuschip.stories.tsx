import { Story, Meta } from "@storybook/react/types-6-0";
import { Statuschip, StatuschipProps } from "./Statuschip";
import { ChipProps } from "@material-ui/core";

export default {
  title: "Statuschip",
  component: Statuschip,
  description: `A Statuschip.`,
  argTypes: {
    size: { control: "text" },
    label: { control: "text" },
    variant: { control: "text" },
  },
} as Meta;

//üëá We create a ‚Äútemplate‚Äù of how args map to rendering
const Template: Story<StatuschipProps & ChipProps> = (args) => (
  <Statuschip {...args} />
);

//üëá Each story then reuses that template
export const CONNECTED = Template.bind({});
CONNECTED.args = {
  size: "small",
  label: "CONNECTED",
  variant: "outlined",
};

export const IN_PROGESS = Template.bind({});
IN_PROGESS.args = {
  size: "small",
  label: "IN_PROGESS",
  variant: "outlined",
};

export const COMPLETED = Template.bind({});
COMPLETED.args = {
  size: "small",
  label: "COMPLETED",
  variant: "outlined",
};

export const CANCELLED = Template.bind({});
CANCELLED.args = {
  size: "small",
  label: "CANCELLED",
  variant: "outlined",
};