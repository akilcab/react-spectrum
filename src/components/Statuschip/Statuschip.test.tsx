import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Statuschip } from "./Statuschip";

describe("<Statuschip />", () => {
  test("rend correctement", () => {
    const component = renderer
      .create(<Statuschip size="small" label="DONE" variant="outlined" />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test("should have content texts in the component", () => {
    const component = render(
      <Statuschip size="small" label="DONE" variant="outlined" />
    );
    expect(component.container).toHaveTextContent("DONE");
    expect(component.container).not.toHaveTextContent("NO_DONE");
  });
});
