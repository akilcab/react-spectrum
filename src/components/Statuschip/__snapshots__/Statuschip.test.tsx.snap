// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Statuschip /> rend correctement 1`] = `
<div
  className="MuiChip-root WithStyles(ForwardRef(Chip))-root-1 WithStyles(ForwardRef(Chip))-root-2 MuiChip-outlined MuiChip-sizeSmall"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
>
  <span
    className="MuiChip-label MuiChip-labelSmall"
  >
    DONE
  </span>
</div>
`;
